py/emitglue.h

```diff
@@ -63,6 +83,14 @@ typedef struct _mp_raw_code_t {
     size_t fun_data_len;
     uint16_t n_obj;
     uint16_t n_raw_code;
+    #if MICROPY_PY_SYS_TRACE
+    mp_bytecode_prelude_t prelude;
```

Comment:
Why does this prelude need to be store explicitly?  Is it just for ease-of-access of the state elements (instead of decoding it all the time)?

Reply:
The pointer (line_info) to the *line number info* section is needed in every callback invocation and every frame change. Also (qstr_source_file) *source_file* and (qstr_block_name) *simple_name* are often needed. Decoding it all the time is quite a slowdown. Especially for LINE events and in the future for INSTR events. Also in the future it may be desirable to precompute all bytecode metada (*line number info*, etc.) in to some faster read format. But for now I think the current tradeoff is OK.